components/Tabs/Tabs.js

```diff
@@ -1,38 +1,43 @@
 class TabLink {
   constructor(element){
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
+    this.info = this.element.dataset.info;
+
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = `data-set=${this.info}` ; 
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
+
+    this.cardElement = this.element.querySelector(`.card[data-set=${this.info}`)
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+    if('all'){
+      this.cards = this.element.querySelectorAll('.card');
     } else {
-      this.cards = ;
+      this.cards = this.element.querySelector(`.card[data-set=${this.info}`);
     }
 
     // Map over the cards array and convert each card element into a new instance of the TabCard class. 
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+   console.log(this.cards);
+    this.cards = Array.from(this.cards).map(singleCard => new TabCard(singleCard));
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener('click',selectTab());
```

Comment:
You can't simply call a method like this, it must binded. How can you bind this method?